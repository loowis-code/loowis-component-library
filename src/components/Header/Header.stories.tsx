import type { Meta, StoryObj } from "@storybook/react";
import Header from "./Header";

const meta: Meta<typeof Header> = {
  title: "Components/Header",
  component: Header,
};

export default meta;

type Story = StoryObj<typeof Header>;

export const photographyWebsite: Story = {
    args: {
        navTabs: ['ALL IMAGES', 'COLLECTIONS', 'IMAGE MAP'],
        navLinks: ['/all-images', '/collections', '/image-map'],
    }
};

export const portfolioWebsite: Story = {
    args: {
        navTabs: ['HOME', 'ABOUT ME', 'PROJECTS'],
        navLinks: ['', '#about-me', '#image-map'],
    }
};
