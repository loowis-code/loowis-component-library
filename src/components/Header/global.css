@font-face {
    font-family: josefin-regular;
    src: url("../../../shared/josefin/JosefinSans-Regular.ttf");
}

.headerContainer {
    padding: 1rem;
    text-decoration: none;
    display: flex;
    justify-content: space-between;
    width: 75%;
    margin: 0 auto;
    border: black 0.25rem solid;
    border-top: none;
    font-family: josefin-regular;
    box-sizing: border-box;
}

.name {
    display: flex;
    cursor: default;
}

.letter {
    width: 4rem;
    height: 4rem;
    padding: 0.25rem;
    margin: 0.25rem;
    justify-content: center;
    align-items: center;
    display: flex;
    color: whitesmoke;
    background-color: #d95d5d;
    border: black 0.25rem solid;
    font-weight: bold;
    font-size: 1.25rem;
    box-shadow: 0.25rem 0.25rem black;
    text-shadow: 0.1rem 0.1rem #000;  
    text-decoration: none;
}

.letter:hover {
    animation: 0.5s shiver;
    text-decoration: none;
    color: whitesmoke;
}

.navTabs {
    display: flex;
    align-items: center;
}

@keyframes shiver {
    0%, 100% { transform: scale(1, 1); }
    25% { transform: scale(0.9, 1.1); }
    50% { transform: scale(1.1, 0.9); }
    75% { transform: scale(0.95, 1.05); }
}

@media (width <= 1320px) {
    .headerContainer {
        width: 100%;
    }
}

@media (width <= 1000px) {
    .headerContainer {
        display: block;
    }

    .navTabs {
        justify-content: center;
    }

    .navTab {
        font-size: smaller;
        aspect-ratio: 2/1;
    }

    .name {
        justify-content: center;
        width: fit-content;
        max-width: 100%;
        justify-content: center;
        align-items: center;
        text-align: center;
        border: #000 0.25rem solid;
        box-shadow: 0.25rem 0.25rem black;
        margin: auto;
    }

    .letter {
        aspect-ratio: 1;
        height: auto;
        padding: 0;
        margin: 0;
        justify-content: center;
        align-items: center;
        display: flex;
        color: whitesmoke;
        background-color: #e81416;
        border: none;
        font-weight: bold;
        font-size: 1.5rem;
        box-shadow: none;
        text-shadow: 0.1rem 0.1rem #000;  
    }
    
    .letter:hover {
        animation: none;
    }
}