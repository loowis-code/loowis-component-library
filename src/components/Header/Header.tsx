import Link from 'next/link';
import { useEffect } from 'react';
import './global.css';
import Button from '../Button';

type HeaderProps = {
    navTabs: string[];
    navLinks: string[];
};

export default function Header(props: HeaderProps) {
    useEffect(() => {
        let name = document.getElementById('name');
        if (!name) return;
        let letters = Array.from(name.children);
        let colors = ['#d95d5d', '#db8525', '#e8c43c', '#bed649', '#9ecbdb', '#c771a1'];
        letters.forEach((letter, index) => {
            (letter as HTMLElement).style.backgroundColor = colors[index];
        });

        let navTabs = document.getElementsByClassName('buttonLink');
        for (let i = 0; i < navTabs.length; i++) {
            (navTabs[i] as HTMLElement).style.setProperty('--buttonBoxShadowHover', colors[i]);
        }
    }, []); 

    return (
        <div className='headerContainer' data-testid='header'>
            <h1 className='name' id='name'>
                <Link className='letter' data-testid='letter' href='/'>L</Link>
                <Link className='letter' data-testid='letter' href='/'>O</Link>
                <Link className='letter' data-testid='letter' href='/'>O</Link>
                <Link className='letter' data-testid='letter' href='/'>W</Link>
                <Link className='letter' data-testid='letter' href='/'>I</Link>
                <Link className='letter' data-testid='letter' href='/'>S</Link>
            </h1>

            <nav className='navTabs'>
                {props.navTabs.map((tab, index) => (
                    <Button key={index} buttonLink={props.navLinks[index]} buttonText={tab}/>
                ))}
            </nav>
        </div>
    );
}