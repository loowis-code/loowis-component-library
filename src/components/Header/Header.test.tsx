import Header from './Header';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';

describe('Header', () => {
    it('renders correctly', () => {
        const tree = render(<Header navTabs={['testString']} navLinks={['/testString']}/>);
        expect(tree).toMatchSnapshot();
      });

    it('should render the header with expected elements', () => {
        const { getByTestId, getAllByTestId, getByText } = render(<Header navTabs={['testString']} navLinks={['/testString']}/>);
        expect(getByTestId('header')).toBeInTheDocument();
        expect(getAllByTestId('letter').length).toEqual(6);
        expect(getByText('testString')).toBeInTheDocument();
    });

    it('should render the header with the correct number of tabs', () => {
        const { getByTestId, getAllByTestId, getByText } = render(<Header navTabs={['testString1', 'testString2']} navLinks={['/testString1', '/testString2']}/>);
        expect(getByTestId('header')).toBeInTheDocument();
        expect(getAllByTestId('navTab').length).toEqual(2);
        expect(getByText('testString1')).toBeInTheDocument();
        expect(getByText('testString2')).toBeInTheDocument();
    });
});