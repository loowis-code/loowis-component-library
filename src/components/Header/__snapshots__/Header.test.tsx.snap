// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="headerContainer"
        data-testid="header"
      >
        <h1
          class="name"
          id="name"
        >
          <a
            class="letter"
            data-testid="letter"
            href="/"
            style="background-color: rgb(217, 93, 93);"
          >
            L
          </a>
          <a
            class="letter"
            data-testid="letter"
            href="/"
            style="background-color: rgb(219, 133, 37);"
          >
            O
          </a>
          <a
            class="letter"
            data-testid="letter"
            href="/"
            style="background-color: rgb(232, 196, 60);"
          >
            O
          </a>
          <a
            class="letter"
            data-testid="letter"
            href="/"
            style="background-color: rgb(190, 214, 73);"
          >
            W
          </a>
          <a
            class="letter"
            data-testid="letter"
            href="/"
            style="background-color: rgb(158, 203, 219);"
          >
            I
          </a>
          <a
            class="letter"
            data-testid="letter"
            href="/"
            style="background-color: rgb(199, 113, 161);"
          >
            S
          </a>
        </h1>
        <nav
          class="navTabs"
        >
          <a
            class="buttonLink"
            data-testid="button"
            href="/testString"
          >
            testString
          </a>
        </nav>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="headerContainer"
      data-testid="header"
    >
      <h1
        class="name"
        id="name"
      >
        <a
          class="letter"
          data-testid="letter"
          href="/"
          style="background-color: rgb(217, 93, 93);"
        >
          L
        </a>
        <a
          class="letter"
          data-testid="letter"
          href="/"
          style="background-color: rgb(219, 133, 37);"
        >
          O
        </a>
        <a
          class="letter"
          data-testid="letter"
          href="/"
          style="background-color: rgb(232, 196, 60);"
        >
          O
        </a>
        <a
          class="letter"
          data-testid="letter"
          href="/"
          style="background-color: rgb(190, 214, 73);"
        >
          W
        </a>
        <a
          class="letter"
          data-testid="letter"
          href="/"
          style="background-color: rgb(158, 203, 219);"
        >
          I
        </a>
        <a
          class="letter"
          data-testid="letter"
          href="/"
          style="background-color: rgb(199, 113, 161);"
        >
          S
        </a>
      </h1>
      <nav
        class="navTabs"
      >
        <a
          class="buttonLink"
          data-testid="button"
          href="/testString"
        >
          testString
        </a>
      </nav>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
